pennylane/compiler/__init__.py

```diff
@@ -44,6 +44,8 @@
     :toctree: api
 
     ~qjit
+    ~for_loop
+    ~while_loop
```

Comment:
Potentially fine as-is, but we could update lines 40-41 as it only talks about `qjit`

Reply:
These are QJIT specific methods and we may need to define a separate `autosummary` for them. Let's get back to this when merging these PRs in PR #4692 (before merging to the main branch).

Reply:
Fixed [c2df222](https://github.com/PennyLaneAI/pennylane/pull/4698/commits/c2df22226b69f4397f1546e4d4115344875c834a)